### PR TITLE
Add Single horizon measurement

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

And use it in the shape control system. This is similar to the BothHorizons measurement, except it is only for one horizon.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

Depends on and includes #4728.

### CODE REVIEW

File: src/ControlSystem/Measurements/SingleHorizon.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}  // namespace Parallel
+namespace domain::Tags {
+template <size_t Dim>
+struct Mesh;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace control_system::measurements {
+template <::domain::ObjectLabel Horizon>
+struct SingleHorizon : tt::ConformsTo<protocols::Measurement> {
```
Comment (nilsdeppe) [inline src/ControlSystem/Measurements/SingleHorizon.hpp:51]: Add some dox just so it shows up?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, please rebase and squash in the one requested change.

File: src/ControlSystem/Measurements/SingleHorizon.hpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  struct SubMeasurement : tt::ConformsTo<protocols::Submeasurement> {
```
Comment (wthrowe) [inline src/ControlSystem/Measurements/SingleHorizon.hpp]: [optional] Your capitalization here is kind of weird.

**Comment by knelli2** [comment]
Squashed both of the changes and also added some docs to the `BothHorizons` measurement while I was at it.

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**